File: 卡码/代码随想录/Python/贪心/02-中等/LeetCode_0135.py
#135. 分发糖果
from typing import *

class Solution:
  def candy(self, ratings: List[int]) -> int:
    can:List[int]=[1]*len(ratings)
    for i in range(1,len(ratings)):
      if ratings[i]>ratings[i-1]:
        can[i]=can[i-1]+1
    for i in range(len(ratings)-2,-1,-1):
      if ratings[i]>ratings[i+1]:
        can[i]=max(can[i+1]+1,can[i])
    return sum(can)

File: 卡码/代码随想录/Go/贪心/02-中等/LeetCode_0135.go
// 135. 分发糖果
package main

import "math"

func candy(ratings []int) int {
	can:=make([]int,len(ratings))
	for i:=0;i<len(can);i++ {
		can[i]=1
	}
	for i:=1;i<len(can);i++ {
		if ratings[i]>ratings[i-1] {
			can[i]=can[i-1]+1
		}
	}
	for i:=len(ratings)-2;i>=0;i-- {
		if ratings[i]>ratings[i+1] {
			can[i]=int(math.Max(float64(can[i+1]+1),float64(can[i])))
		}
	}
	sum:=0
	for _,val:=range can {
		sum+=val
	}
	return sum
}
